cli/create-spacetime/src/index.ts

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env node
+
+import { Command } from "commander";
+import inquirer from "inquirer";
+import chalk from "chalk";
+import fs from "fs-extra";
+import path from "path";
+import { sync } from "cross-spawn";
+
+import { createProject } from "./lib/create-project.js";
+import {
+  getTemplateChoices,
+  isValidTemplate,
+  TEMPLATES,
+  DEFAULT_TEMPLATE,
+} from "./templates/index.js";
+import { detectPackageManager, getPackageVersion } from "./utils/package.js";
+import { getValidPackageName, checkDependencies } from "./utils/validation.js";
+import { formatTargetDir, isEmpty, emptyDir } from "./utils/directory.js";
+import { checkSpacetimeLogin } from "./utils/spacetime.js";
+import { DEFAULT_TARGET_DIR } from "./constants.js";
+
+const cwd = process.cwd();
+
+async function handleCreateProject(projectNameArg: string, options: any) {
+  console.log(chalk.blue("Welcome to create-spacetime!"));
+  console.log();
+
+  const packageManager = detectPackageManager();
+  console.log(`Using package manager: ${chalk.green(packageManager)}`);
+  console.log();
+
+  const deps = await checkDependencies();
+
+  if (!deps.node || !deps.npm) {
+    console.error(chalk.red("Error: Node.js and npm are required but not found."));
+    console.error(`Please install Node.js from ${chalk.bold("https://nodejs.org/")}`);
+    process.exit(1);
+  }
+
+  if (!deps.spacetime) {
```

Comment:
Is there a reason we couldn't just install this for them?

Reply:
Eventually I think we'd like to move most of this functionality directly into the `spacetime` cli command, so it would make sense that this actually just becomes a simple bootstrapping TypeScript script to install the CLI and then run a command like:

`spacetime init --template foobar/baz`